testcases/PowerNVDump.py

```diff
@@ -1324,6 +1334,293 @@ def runTest(self):
                 self.fail("KernelArgTest failed to update kernel args")
 
 
+class OpTestWatchdog(PowerNVDump):
+    '''
+    This test verifies "watchdog module" with diffrent scenarios like
+    1. Watchdog module load and unload.
+    2. watchdog with action 1/ Reboot LPAR.
+    3. watchdog with action 0/ Halt LPAR.
+    4. watchdog with action 2 /dump collect
+                a. dump collect to a local disk
+                b. dump collect to a NFS disk
+                c. dump collect to a SAN FC disk
+    '''
+
+    def setUp(self):
+        super(OpTestWatchdog, self).setUp()
+        self.mg_system = self.cv_HMC.mg_system
+
+        if not self.cv_HMC.is_lpar_in_managed_system(self.mg_system, self.cv_HMC.lpar_name):
+            raise OpTestError("Lpar %s not found in managed system %s" % (
+                              self.cv_HMC.lpar_name, self.mg_system))
+
+    def get_watchdog_tool(self):
+        '''
+        This funtions copies the watchdog ".c" file from
+        test_binaries to root directory and compiles and
+        creates "watchdog-countdown" tool directory,using
+        this can trigger the watchdog events.
+        '''
+        filename = "watchdog-countdown.c"
+        self.cv_HOST.copy_test_file_to_host(filename, dstdir="/root")
+        base_file = os.path.splitext(filename)[0]
+        if not self.cv_HOST.host_run_command(f"make {base_file}"):
```

Comment:
Does host_run_command() internally create a makefile? IF not how will this work without makefile? Don't we need to use gcc?

Reply:
Nice catch,  
```
abdul:~$ make watch.c
make: Nothing to be done for 'watch.c'.
abdul:~$ echo $?
0
```
not sure how does the test still passed, as binary is not created.. better to use gcc